lib/java/com/google/android/material/lists/MaterialListAdapter.java

```diff
@@ -0,0 +1,54 @@
+package com.google.android.material.lists;
+
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.lang.reflect.InvocationTargetException;
+
+/**
+ * An adapter that uses {@link MaterialViewHolder}s and binds data to them using {@link MaterialListBinder}
+ */
+public class MaterialListAdapter<T, E extends MaterialViewHolder> extends RecyclerView.Adapter<E> {
+
+  private MaterialListBinder<T, E> materialListBinder;
+  private Class<E> viewHolderClass;
+
+  public MaterialListAdapter(MaterialListBinder<T, E> materialListBinder, Class<E> viewHolderClass) {
+    this.materialListBinder = materialListBinder;
+    this.viewHolderClass = viewHolderClass;
+  }
+
+  @NonNull
+  @Override
+  public E onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+
+    try {
+      return viewHolderClass.getConstructor(ViewGroup.class).newInstance(parent);
```

Comment:
The reflection implemention is safe, and it does create view holders appropriately. I am going to design the adapter so that you can override onCreate but with the ViewHolder already created (and onBind).